My project/Assets/Scripts/CreatureLogic/Actions/Breeding/LookForMate.cs

```diff
@@ -0,0 +1,108 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class LookForMate : IAction
```

Comment:
this whole class seems kind of redundant, as above in Breed, we check if a target is either in Breed already or LookForMate. can we consolidate them into one action - specifically, just use Breed for both of these usecases?

Reply:
I agree, this problem also happens with find food and eat food, so to fix this I believe its better to finally revamp the creature data so things like findfood and eatfood share the same food variable and findmate and breed have the same target as was discussed earlier, I'll change this in the next update